### PR TITLE
plotter: introduce a special Histogram for Log-y plots

### PR DESCRIPTION
**Pull request by sbinet**
Plotting a histogram with a Y-log scale needs to carefully take into
account the low bar of a histogram bin.
The bottom of a histogram bin is usually 0, a problematic value for a
log-y scale.

This CL introduces plotter.LogHistogram, a light specialization of
plotter.Histogram.

Fixes gonum/plot#377.

Please take a look.

### CODE REVIEW

File: plotter/histogram.go [hunk 1]
```diff
@@ -193,3 +193,55 @@ type HistogramBin struct {
...
+		}
+		if h.FillColor != nil {
+			c.FillPolygon(h.FillColor, c.ClipPolygonXY(pts))
+		}
+		pts = append(pts, vg.Point{X: trX(bin.Min), Y: trY(ymin)})
+		c.StrokeLines(h.LineStyle, c.ClipLinesXY(pts)...)
+	}
+}
+
+// DataRange returns the minimum and maximum X and Y values
```
Comment (ctessum) [inline plotter/histogram.go]: period at end of sentence

File: plotter/histogram.go [hunk 2]
```diff
@@ -193,3 +193,55 @@ type HistogramBin struct {
...
+			xmin = bin.Min
+		}
+		if bin.Weight < ymin {
+			ymin = bin.Weight
+		}
+		if bin.Weight > ymax {
+			ymax = bin.Weight
+		}
+	}
+	ymin /= 2.5
```
Comment (ctessum) [inline plotter/histogram.go]: Why 2.5 here?

File: plotter/histogram.go (hunk 1 above)
Comment (ctessum) [inline plotter/histogram.go]: Period at end of sentence

File: plotter/histogram_test.go [hunk 3]
```diff
@@ -59,6 +59,40 @@ func ExampleHistogram() {
...
+	}
+	p.Title.Text = "Histogram in logy"
+	p.Y.Scale = plot.LogScale{}
+	p.Y.Tick.Marker = plot.LogTicks{}
+
+	h, err := NewHist(vals, 16)
+	if err != nil {
+		log.Fatal(err)
+	}
+	// h.YMin = 0.1
```
Comment (ctessum) [inline plotter/histogram_test.go]: remove comment

File: plotter/histogram_test.go [hunk 4]
```diff
@@ -59,6 +59,40 @@ func ExampleHistogram() {
...
+	vals := make(Values, n)
+	for i := 0; i < n; i++ {
+		vals[i] = rnd.NormFloat64()
+	}
+
+	p, err := plot.New()
+	if err != nil {
+		log.Panic(err)
+	}
+	p.Title.Text = "Histogram in logy"
```
Comment (ctessum) [inline plotter/histogram_test.go]: maybe "log Y"?

File: plotter/histogram.go (hunk 1 above)
Comment (sbinet) [reply plotter/histogram.go]: done

File: plotter/histogram.go (hunk 2 above)
Comment (sbinet) [reply plotter/histogram.go]: added some terse explanation

File: plotter/histogram_test.go (hunk 4 above)
Comment (sbinet) [reply plotter/histogram_test.go]: done

File: plotter/histogram_test.go (hunk 3 above)
Comment (sbinet) [reply plotter/histogram_test.go]: done

**Comment by sbinet** [comment]
abandoned in favour of https://github.com/gonum/plot/pull/493

**Pull request closed without merging**